modin/engines/base/frame/data.py

```diff
@@ -329,7 +329,7 @@ def _filter_empties(self):
         self._column_widths_cache = [w for w in self._column_widths if w != 0]
         self._row_lengths_cache = [r for r in self._row_lengths if r != 0]
 
-    def _apply_index_objs(self, axis=None):
+    def synchronize_labels(self, axis=None):
         """
         Apply the index object for specific `axis` to the `self._partitions` lazily.
```

Comment:
I feel that after changing the name the summary of this method needs to be updated, too.

Reply:
Yes, we can update it a bit.